applications/ShapeOptimizationApplication/python_scripts/value_logger_base.py

```diff
@@ -72,58 +87,46 @@ def __LogValuesToHistory( self, additional_values_dictionary ):
 
     # --------------------------------------------------------------------------
     def __LogObjectiveValuesToHistory( self ):
-        objective_id = self.specified_objectives[0]["identifier"].GetString()
+        objective_id = self.objectives[0]["identifier"].GetString()
+        is_first_log = self.history["value"][objective_id] == {}
```

Comment:
```suggestion
        is_first_log = len(self.history["value"][objective_id]) == 0
```

Reply:
Is there a reason why not to use the rather explicit "== {}". I mean, is this bad practice?

Reply:
Typically you find "len()==0" or even "not my_dict". Nothing wrong "== {}" its just not so common.